extension/persistence/relational-jdbc/src/main/java/org/apache/polaris/extension/persistence/relational/jdbc/QueryGenerator.java

```diff
@@ -190,6 +210,30 @@ public static String generateWhereClause(@Nonnull Map<String, Object> whereClaus
     return !whereConditionsString.isEmpty() ? (" WHERE " + whereConditionsString) : "";
   }
 
+  @VisibleForTesting
+  public static String generateVersionQuery() {
+    return "SELECT version_value FROM POLARIS_SCHEMA.VERSION";
+  }
+
+  @VisibleForTesting
+  public static String generateOverlapQuery(String realmId, long parentId, String location) {
+    String[] components = location.split("/");
+    StringBuilder locationClauseBuilder = new StringBuilder();
+    StringBuilder pathBuilder = new StringBuilder();
+    for (String component : components) {
+      pathBuilder.append(component).append("/");
+      locationClauseBuilder.append(String.format(" OR location = '%s'", pathBuilder));
+    }
+    locationClauseBuilder.append(String.format(" OR location LIKE '%s%%'", location));
+    String query = "SELECT " + String.join(", ", new ModelEntity().toMap().keySet());
+
+    // TODO harden against realmId in this method and others
+    return query
+        + String.format(
+            " FROM POLARIS_SCHEMA.entities WHERE realm_id = '%s' AND parent_id = %d AND (1 = 2%s)",
```

Comment:
would prefer if we can move infering `POLARIS_SCHEMA.entitites` to the common logic below.